website/guides/running-terraform-in-automation.html.md

```diff
@@ -0,0 +1,279 @@
+
+Alternatively, the automation tool can set the environment variable
+`TF_WORKSPACE` to an existing workspace name, which overrides any selection
+made with the `terraform workspace select` command. Using this environment
+variable is recommended only for non-interactive usage, since in a local shell
+environment it can be easy to forget the variable is set and apply changes
+to the wrong state.
+
+In some more complex situations it is impossible to share the same backend
+configuration across environments. For example, the environments may exist
+in entirely separate accounts within the target service, and thus need to
+use different credentials or endpoints for the backend itself. In such
+situations, backend configuration settings can be overridden via
+[the `-backend-config` option to `terraform init`](/docs/commands/init.html#backend-config).
+
+## Pre-installed Plugins
+
+In default usage, [`terraform init`](/docs/commands/init.html#backend-config)
+downloads and installs the plugins for any providers used in the configuration
+automatically, placing them in a subdirectory of the `.terraform` directory.
+This affords a simpler workflow for straightforward cases, and allows each
+configuration to potentially use different versions of plugins.
+
+In automation environments, it can be desirable to disable this behavior
+and instead provide a fixed set of plugins already installed on the system
+where Terraform is running. This then avoids the overhead of re-downloading
+the plugins on each execution, and allows the system administrator to control
+which plugins are available.
+
+To use this mechanism, create a directory somewhere on the system where
+Terraform will run and place into it the plugin executable files. The
+plugin release archives are available for download on
+[releases.hashicorp.com](https://releases.hashicorp.com/). Be sure to
+download the appropriate archive for the target operating system and
+architecture.
+
+After extracting the necessary plugins, the contents of the new plugin
+directory will look something like this:
+
+```
+$ ls -lah /usr/lib/custom-terraform-plugins
+-rwxrwxr-x 1 user user  84M Jun 13 15:13 terraform-provider-aws-v1.0.0-x3
+-rwxrwxr-x 1 user user  84M Jun 13 15:15 terraform-provider-rundeck-v2.3.0-x3
+-rwxrwxr-x 1 user user  84M Jun 13 15:15 terraform-provider-mysql-v1.2.0-x3
+```
+
+The version information at the end of the filenames is important so that
+Terraform can infer the version number of each plugin. It is allowed to
+concurrently install multiple versions of the same provider plugin,
+which will then be used to satisfy
+[provider version constraints](/docs/configuration/providers.html#provider-versions)
+from Terraform configurations.
+
+With this directory populated, the usual auto-download and discovery behavior
+can be bypassed using the `-plugin-dir` option to `terraform init`:
+
+* `terraform init -input=false -plugin-dir=/usr/lib/custom-terraform-plugins`
```

Comment:
`-plugin-dir` actually circumvents all that in favor of explicitly naming all of the locations where plugins are, with the assumption being that if you're trying to exercise central control then you don't want other plugins to "sneak in" from other directories you're not thinking about.

I'll revise here to try to make that fact more obvious. Right now "the usual discovery behavior" is trying to refer to that, but indeed it would be helpful to link somewhere to describe what the "usual" behavior is.